### PR TITLE
#186 Make standard predictor predict from the day before start_date

### PR DESCRIPTION
**Pull request by ofrancon**
Made standard predictor predict from the day before start_date. We had 2 issues:
1. The class level DataFrame containing the Oxford data was the one we trained on (Jan 11)
2. We were predicting from 2 days before start date instead of from the day before start date. For instance, if start date is July 11, we were predicting from the number of cases on July 9 instead of on July 10. 

Fixes:
For 1., now truncating the Oxford DataFrame to start date, excluded. Also not storing this DataFrame at the class level anymore to avoid messing up potential future calls to predict(...). 
Issue 2. was caused by the fact we don't know the outcome (y) for the last day when training, so we exclude it. But when predicting we should include it because we don't care about the y.

### CODE REVIEW

File: covid_xprize/standard_predictor/xprize_predictor.py [hunk 1]
```diff
@@ -89,8 +89,6 @@ def __init__(self, path_to_model_weights, data_url):
                     f"Data file not found at {DATA_FILE_PATH}"
 
         self.df = self._prepare_dataframe(data_url)
-        geos = self.df.GeoID.unique()
-        self.country_samples = self._create_country_samples(self.df, geos)
```
Comment (ofrancon) [inline covid_xprize/standard_predictor/xprize_predictor.py:93]: `country_samples` is not a class level variable anymore

File: covid_xprize/standard_predictor/xprize_predictor.py [hunk 2]
```diff
@@ -109,10 +107,15 @@ def predict(self,
                     "Date": [],
                     "PredictedDailyNewCases": []}
 
+        # Fix for past predictions
+        geos = npis_df.GeoID.unique()
+        truncated_df = self.df[self.df.Date < start_date_str]
+        country_samples = self._create_country_samples(truncated_df, geos, False)
```
Comment (ofrancon) [inline covid_xprize/standard_predictor/xprize_predictor.py:113]: Instead we recreate `country_samples` when `predict` is called, according to the Oxford DataFrame truncated to before the start date.

File: covid_xprize/standard_predictor/xprize_predictor.py [hunk 3]
```diff
@@ -284,11 +287,12 @@ def _load_additional_context_df():
         return additional_context_df
 
     @staticmethod
-    def _create_country_samples(df: pd.DataFrame, geos: list) -> dict:
+    def _create_country_samples(df: pd.DataFrame, geos: list, is_training: bool) -> dict:
```
Comment (ofrancon) [inline covid_xprize/standard_predictor/xprize_predictor.py:290]: Now distinguishing between `training` and `predicting` modes

File: covid_xprize/standard_predictor/xprize_predictor.py [hunk 4]
```diff
@@ -300,30 +304,35 @@ def _create_country_samples(df: pd.DataFrame, geos: list) -> dict:
...
             context_samples = []
             action_samples = []
-            outcome_samples = []
-            nb_total_days = outcome_data.shape[0]
+            if is_training:
+                outcome_data = np.array(cdf[outcome_column])
+                outcome_samples = []
+                nb_total_days = context_data.shape[0]
+            else:
+                nb_total_days = context_data.shape[0] + 1
```
Comment (ofrancon) [inline covid_xprize/standard_predictor/xprize_predictor.py:314]: When predicting, we do not need the outcomes (y) anymore, and we want to include the last day of context and action samples.

**Review by EKMeyerson — APPROVED** [review]
lgtm; thanks

**Pull request merged**